pyccel/ast/type_annotations.py

```diff
@@ -254,7 +237,8 @@ class UnionTypeAnnotation(PyccelAstNode):
 
     def __init__(self, *type_annotations):
         annots = [ti for t in type_annotations for ti in (t.type_list if isinstance(t, UnionTypeAnnotation) else [t])]
-        self._type_annotations = tuple(set(annots))
+        # Strip out repeats
+        self._type_annotations = tuple({a: None for a in annots}.keys())
```

Comment:
You went from a set to a dictionary in order to preserve the order?

Reply:
Yes. With a set the code will be different every time it is printed as the annotations are reordered